src/unix/mmap.c

```diff
@@ -121,14 +124,17 @@ boolean do_demand_page(u64 vaddr, vmap vm, context frame)
                of a syscall (under the kernel lock). As such, we are free to set up an asynchronous page
                fill, allocate memory, etc. */
             assert(!faulting_kernel_context);
+            assert(this_cpu_has_kernel_lock());
             kernel_demand_page_completed = false;
             pagecache_map_page(vm->cache_node, node_offset, page_addr, flags,
-                               (status_handler)&do_kernel_demand_pf_complete);
+                               (status_handler)&do_kernel_demand_pf_complete,
+                               true /* complete on bhqueue */);
             if (kernel_demand_page_completed) {
                 pf_debug("   immediate completion\n");
                 return true;
             }
             faulting_kernel_context = suspend_kernel_context();
+            current_cpu()->have_kernel_lock = false;
```

Comment:
This is redundant with suspend_kernel_context, which also clears have_kernel_lock.

Reply:
Ok, I removed the one in suspend_kernel_context for consistency (as kernel_frame_return sets it above).